mesg/deploy_deployer.go

```diff
@@ -0,0 +1,135 @@
+package mesg
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/logrusorgru/aurora"
+	"github.com/mesg-foundation/core/database/services"
+	"github.com/mesg-foundation/core/service"
+	"github.com/mesg-foundation/core/service/importer"
+	uuid "github.com/satori/go.uuid"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type serviceDeployer struct {
+	Statuses chan string
+
+	mesg *MESG
+}
+
+func newServiceDeployer(mesg *MESG) *serviceDeployer {
+	return &serviceDeployer{
+		mesg: mesg,
+	}
+}
+
+func (d *serviceDeployer) FromGitURL(url string) (*service.Service, *importer.ValidationError, error) {
+	d.sendStatus("Downloading service...")
+	path, err := d.createTempDir()
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := d.gitClone(url, path); err != nil {
+		return nil, nil, err
+	}
+	return d.deploy(path)
+}
+
+func (d *serviceDeployer) gitClone(repoURL string, path string) error {
```

Comment:
I think gitClone should be moved to some utils pkg (maybe xgit?) but it's not part of service Deploeyr

Reply:
+ missing docs

Reply:
i see that we have quite a lot _utils.go_, _utils_ package inside the project. i think we should reduce them and serve under a root level utils package as possible but use different package names for each *different* util under _utils/_ e.g.:

```
util/
  gitutil/ > gitutil.Clone()
```

lets also give a second check to this article about package namings: https://blog.golang.org/package-names

Reply:
docs are done